bigquery/storage/managedwriter/managed_stream.go

```diff
@@ -293,11 +294,17 @@ func (ms *ManagedStream) Close() error {
 }
 
 // AppendRows sends the append requests to the service, and returns one AppendResult per row.
-func (ms *ManagedStream) AppendRows(data [][]byte, offset int64) ([]*AppendResult, error) {
+func (ms *ManagedStream) AppendRows(ctx context.Context, data [][]byte, offset int64) ([]*AppendResult, error) {
 	pw := newPendingWrite(data, offset)
+	// check flow control
+	if err := ms.fc.acquire(ctx, pw.reqSize); err != nil {
+		// in this case, we didn't acquire, so don't pass the flow controller reference to avoid a release.
+		pw.markDone(NoStreamOffset, err, nil)
+	}
```

Comment:
@shollyman Is there a missing return here? We've observed the following panic, which seems like it could be related to markDone resetting pw.request above before the append call proceeds below.

```
runtime error: invalid memory address or nil pointer dereference
at cloud.google.com/go/bigquery/storage/managedwriter.(*ManagedStream).append ( /go/src/github.com/ScriptRock/webscan-golang/vendor/cloud.google.com/go/bigquery/storage/managedwriter/managed_stream.go:284 )
at cloud.google.com/go/bigquery/storage/managedwriter.(*ManagedStream).AppendRows ( /go/src/github.com/ScriptRock/webscan-golang/vendor/cloud.google.com/go/bigquery/storage/managedwriter/managed_stream.go:363 )
```

Reply:
Created https://github.com/googleapis/google-cloud-go/issues/5428 to follow up on this.  Sheer luck I saw this comment, but looks likely.

Reply:
Thanks!